### ISSUE TITLE
sync wallet failed: state validation error

### ISSUE THREAD

**Issue by ghost** [themeliolabs/melwalletd#18]
This bug was reported by `Venkat#6928` in the discord community, and I compiled the information he gave me and submitted the issue.

## Description

When running melwalletd, a strange error occurs, which is logged as:

```shell
[2023-XX-XXTXX:XX:XXZ INFO melwalletd] using node RPC 185.132.179.104:41814
[2023-XX-XXTXX:XX:XXZ INFO melwalletd] starting RPC server at 127.0.0.1:11773
[2023-XX-XXTXX:XX:XXZ WARN melwalletd::database] VERY behind, so doing a full sync of <WALLET_ADDRESS>
[2023-XX-XXTXX:XX:XXZ WARN melwalletd::state] sync <WALLET_NAME> failed: state validation error: got incomplete list of 52 coins rather than 51
```

Here are some images:

![image](https://user-images.githubusercontent.com/95868829/224413080-a28fbfec-1194-4e2f-852e-2604d810a141.png)

![image](https://user-images.githubusercontent.com/95868829/224413099-6e4bafe1-da4e-487b-8dee-0e920f9b2dcd.png)

![image](https://user-images.githubusercontent.com/95868829/224413105-78e989ea-1136-4b7b-bb6b-061816ad4e0a.png)

![image](https://user-images.githubusercontent.com/95868829/224413113-e268facf-7abd-477d-b3a5-d3f87757c0b1.png)

![image](https://user-images.githubusercontent.com/95868829/224413125-73a136f2-eed2-42e9-a3de-8550e9dfaea9.png)

![image](https://user-images.githubusercontent.com/95868829/224413147-5050f916-9933-4e3c-8a16-0b9d6f2e7880.png)

## Environment Information

- OS: Windows/Ubuntu/macOS
- Commands for melwalletd: `melwalletd --wallet-dir <PATH>`
- Commands for melwallet-cli: `melwallet-cli summary -w <WALLET_NAME>`
- melwalletd version: 0.6.3
- melwallet-client version: 0.7.3

**Comment by antidmg** [comment]
Looking into this now -- it's most likely a bug with the indexer / `get_some_coins` RPC

**Comment by antidmg** [comment]
This is most likely fixed in `melblkidx 0.7.2` with a [fix](https://github.com/mel-project/melblkidx/commit/f7969a0613ed963f52bab01175b9aabc4de753ee) for handling liquidity transactions. We will do some more extensive testing on our side, but please let us know if you still see this issue after upgrading in the meantime.

**Comment by ghost** [comment]
This bug has been fixed after melblkidx and melnode updated and rebuilt their indexes.
